framework.py

```diff
@@ -0,0 +1,77 @@
+# Copyright (c) 2019 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import os
+import sys
+import platform
+
+import sgtk
+logger = sgtk.platform.get_logger(__name__)
+
+
+def get_vendors_path():
+    """Return the path to the vendors folder.
+
+    The vendors folder is a way to package the dependecies of this framework in a "It just works" manner.
+
+    :returns: path to the vendors folder
+    :rtype: str
+    """
+    def get_python_version():
+        if sys.hexversion < 0x3000000:
+            return "2"
+
+        return "3"
+
+    def get_platform():
+        return platform.uname()[0]
+
+    return os.path.abspath(os.path.join(
+        os.path.dirname(__file__), "Vendors", get_platform(), get_python_version()))
+
+
+def patch_environment():
+    """
+    This function patch the python path to add the required modules to the python path.
+    """
+    # Try to import websocket to see if we are good to go.
+
+    vendor_path = get_vendors_path()
+
+    if vendor_path not in sys.path:
+        logger.debug("Adding {} to the python path".format(vendor_path))
```

Comment:
I can't say for sure, but I think there are still versions of DCCs we support that have Python 2.6, and in that version you can't use empty braces for formatting. You need to use `{0}`. It seems `websocket_client` also supports 2.6, so I think it's worth giving it a shot.

Reply:
Can't wait for f"strings" :p